distribution/client/test/snapshotter.test.js

```diff
@@ -0,0 +1,48 @@
+const Snapshotter = require('../src/lib/snapshotter');
+const tmp         = require('tmp');
+const fs          = require('fs');
+
+describe('The snapshotting module', () => {
+
+  let tmpDir = '';
+  beforeEach(() => {
+    tmpDir = tmp.dirSync({unsafeCleanup: true});
+  });
+
+  afterEach(() => {
+    tmpDir.removeCallback();
+  });
+
+  describe('init()', () => {
+    it('should init a repo', () => {
+      const snapshotter = new Snapshotter();
+      snapshotter.init(tmpDir.name);
+
+      // Called twice does not crash:
+      // important since this is what will happen
+      // on container/client restart
+      snapshotter.init(tmpDir.name);
+    });
+  });
+  describe('snapshot()', () => {
```

Comment:
There is no test coverage of the `git()` function which is I believe the most risky one in this module.

We're a bit crunched for time, but @batmat would you please file a follow up ticket to add that test coverage for the potential error branches of `git()` into Milestone 2?

Reply:
Done https://issues.jenkins-ci.org/browse/JENKINS-53503

Reply:
note that snapshot() calls git and, so does the constructor, so the underlying is tested, but why not.